tests/maxwell_solver_test.py

```diff
@@ -13,3 +15,109 @@ def test_1d_uniform():
     true_phi = -2 * np.pi * (ms.grid**2 - 1)
     err = np.abs(true_phi - phi)
     np.testing.assert_allclose(err, np.zeros(len(err)), atol=1e-9)
+
+
+def test_2d_A():
+    ms = MaxwellSolver2D()
+    A = ms.set_A(4)
+
+    true_A = np.array([[-4, 1, 1, 0], [1, -4, 0, 1], [1, 0, -4, 1], [0, 1, 1, -4]])
+    np.testing.assert_allclose(A, true_A, atol=1e-9)
+
+
+def test_2d_boundary():
+    pass
+
+
+def test_2d_laplace():
+    """Test 2D Poisson solver for rho = 0. Ex taken from Ex 3.4 in Griffiths E&M, 4th ed"""
+    N = 100
+    V0 = 1
+    a = 2
+    b = 1
+
+    grid = Uniform2DGrid(num_points=N, x_min=-b, x_max=b, y_min=0, y_max=a)
+    boundary_conditions = {
+        "bottom": np.zeros(N),
+        "top": np.zeros(N),
+        "left": V0 * np.ones(N),
+        "right": V0 * np.ones(N),
+    }
+    ms = MaxwellSolver2D(grid=grid, boundary_conditions=boundary_conditions)
+    rho = np.zeros(ms.grid[0].shape)
+
+    phi = ms.solve(rho)
+
+    max_n = 250
+    X, Y = np.meshgrid(np.linspace(-b, b, N), np.linspace(0, a, N))
+    true_phi = np.zeros(X.shape)
+
+    for i in range(max_n):
+        if i % 2 == 1:
+            true_phi += (
+                4
+                * V0
+                / np.pi
+                * 1
+                / i
+                * np.cosh(i * np.pi * X / a)
+                / np.cosh(i * np.pi * b / a)
+                * np.sin(i * np.pi * Y / a)
+            )
+
+    # There are some issues that I believe are related to the convergence of an infinite Fourier series
+    # For now, these "magic" numbers should be kept fixed.
```

Comment:
```suggestion
    # FIXME There are some issues that I believe are related to the convergence of an infinite Fourier series
    # FIXME For now, these "magic" numbers should be kept fixed.
```
Just marking these lines so we can come back and address the issue later.